src/Nethermind/Nethermind.Blockchain/TransactionPools/Storages/PersistentTransactionStorage.cs

```diff
@@ -0,0 +1,55 @@
+using System;
+using Nethermind.Core;
+using Nethermind.Core.Crypto;
+using Nethermind.Core.Encoding;
+using Nethermind.Core.Specs;
+using Nethermind.Dirichlet.Numerics;
+using Nethermind.Store;
+
+namespace Nethermind.Blockchain.TransactionPools.Storages
+{
+    public class PersistentTransactionStorage : ITransactionStorage
+    {
+        private readonly IDb _database;
+        private readonly ISpecProvider _specProvider;
+
+        public PersistentTransactionStorage(IDb database, ISpecProvider specProvider)
+        {
+            _database = database;
+            _specProvider = specProvider;
+        }
+
+        public Transaction Get(Keccak hash)
+        {
+            var transactionData = _database.Get(hash);
+
+            return transactionData == null
+                ? null
+                : Rlp.Decode<Transaction>(new Rlp(transactionData), RlpBehaviors.Storage);
+        }
+
+        public Transaction[] GetAll()
+        {
```

Comment:
exception

Reply:
It uses `IDb` which doesn't have yet a possibility to get all data at once.

Reply:
needs a different db config - you can check with Grzesiek - he had trouble to get it working in rocksdb and used some simple file storage instead...I would not use file storage for this though as the information here is much more important

Reply:
`MultiGet()` wouldn't work?